core/src/main/java/org/apache/lucene/queryparser/classic/MapperQueryParser.java

```diff
@@ -779,7 +757,10 @@ private Query applySlop(Query q, int slop) {
                 builder.add(terms[i], positions[i]);
             }
             pq = builder.build();
-            pq.setBoost(q.getBoost());
+            //nocommit why do we deal with boost here? isn't that what applyBoost does?
```

Comment:
I added two asserts, I haven't made up my mind on whether I prefer the asserts or your code snippets though :)  let me know what you prefer